mne/event.py

```diff
@@ -745,26 +746,38 @@ def make_fixed_length_events(raw, id, start=0, stop=None, duration=1.):
         of the recording.
     duration: float
         The duration to separate events by.
+    first_samp: bool
+        If True (default), times will have raw.first_samp added to them, as
+        in :func:`mne.find_events`. This behavior is not desirable if the
+        returned events will be combined with event times that already
+        have ``raw.first_samp`` added to them, e.g. event times that come
+        from :func:`mne.find_events`.
 
     Returns
     -------
     new_events : array
         The new events.
     """
     start = raw.time_as_index(start)
-    start = start[0] + raw.first_samp
     if stop is not None:
         stop = raw.time_as_index(stop)
+    else:
+        stop = [raw.last_samp + 1]
+    if first_samp:
+        start = start[0] + raw.first_samp
         stop = min([stop[0] + raw.first_samp, raw.last_samp + 1])
```

Comment:
at this point in the code `stop` could be just an `int`. I think that it's better to convert during `time_as_index` as e.g. `stop = raw.time_as_index(stop)[0]` so that you always have `start` and `stop` as `int` (or `None`) instead of either an `int` or `list`, as it is at this point in the code